### PR TITLE
#207 implement RTL support

### PR DESCRIPTION
**Pull request by coffeeandwork**

### CODE REVIEW

**Comment by coffeeandwork** [comment]
@leonardorafael I would like to get your early impression on whether you are good this PR.

**Comment by coffeeandwork** [comment]
Here is RTL broken (before this PR): https://codepen.io/coffeeandwork/pen/yLGMpKb

Here is RTL fixed (after this PR): https://codepen.io/coffeeandwork/pen/QWzpaPv

**Comment by coffeeandwork** [comment]
@leonardorafael any chance you could add me as a contributor to this project?

**Comment by leonardorafael** [comment]
> @leonardorafael any chance you could add me as a contributor to this project?

Sure @coffeeandwork, thanks for help.

Hey @laurentpayot do you want to be a contributor too? You always help this project too.

**Comment by laurentpayot** [comment]
@leonardorafael sure, I’m always glad to help Beercss because Beercss already helps me in my projects :beers:

**Comment by coffeeandwork** [comment]
> @leonardorafael sure, I’m always glad to help Beercss because Beercss already helps me in my projects 🍻

I feel the same way. thank you @leonardorafael

**Comment by KicKerBNU** [comment]
Hello @laurentpayot and @coffeeandwork i just want to say its a pleasure to have you as contributors of the Beercss. 

Thanks for all the work and support of the Beercss community.

**Comment by coffeeandwork** [comment]
Thank you for welcoming us into the community and for such a great project.

**Review by leonardorafael — APPROVED** [review]
👍

**Comment by coffeeandwork** [comment]
@leonardorafael please hold off on merging this PR. I am testing to make sure each of the individual components work well. I will report back with a confirming once I am done. I am trying to shoot for getting that done by tomorrow. I assume you don't mind if I merge the PR tomorrow after I finish this? Also, what are the general guidelines you would like to follow amongst the contributors when it comes to merging PRs. Should we wait for PR approval from you before merging? Or should we work on getting simply any other contributor to review / approve the PR? Or are we OK to use our best judgement and just go for it (this risks bugs, but in the spirit of trying to move the project forward faster maybe we can trust that contributors will test carefully, and that if bugs are introduced, we will be quick to revert the PR that introduced the bug or provide a quick fix? Please let us know your thoughts? Should we also add a section about this to the contributing guidelines?

**Comment by leonardorafael** [comment]
@coffeeandwork  in my opinion we can adopt 3 rules at this moment:

1) All contributors will be add to review a PR (we need at least 1 approve).
2) I would like to merge all PRs at this moment, ok?
3) We can add a label "do not merge" when we are working/reviewing a opened/approved PR (and don't forget to remove the label when done).

I think we can start like this.

**Comment by leonardorafael** [comment]
Also when we will release a new version we will open a PR to merge `dev` into `main`. At this point we need to test all.

**Comment by coffeeandwork** [comment]
> @coffeeandwork in my opinion we can adopt 3 rules at this moment:
> 
> 1. All contributors will be add to review a PR (we need at least 1 approve).
> 2. I would like to merge all PRs at this moment, ok?
> 3. We can add a label "do not merge" when we are working/reviewing a opened/approved PR (and don't forget to remove the label when done).
> 
> I think we can start like this.

Sounds good. Thank you.

**Review by laurentpayot — APPROVED** [review]
LGTM (from what I can tell :sweat_smile:)

**Comment by KicKerBNU** [comment]
@coffeeandwork Hello, im not familiar to RTL so i downloaded an Chrome Extension to make some tests and i have some doubts: 

Badge:

Before RTL update
image

After RTL update
image

Here we can see the position of the badge now making a right to left default ( as expected )

Before RTL update
Badge top left
image

After RTL update
Badge top left
image

Here we can see the name of the classes dont match to the position of the badge
We have a Badge with a class top left on RTL we should expect to have a class called left on the right side ?
This is the expected behavior or its a bug ?

File: src/cdn/elements/pages.css
Comment (KicKerBNU) [inline src/cdn/elements/pages.css:29]: not sure if the left page should be right

File: src/cdn/elements/pages.css
Comment (KicKerBNU) [inline src/cdn/elements/pages.css:37]: not sure right page should be left

File: src/cdn/elements/tables.css
Comment (KicKerBNU) [inline src/cdn/elements/tables.css:22]: tables seems to have the same problem with right and left align

File: src/cdn/elements/tabs.css
Comment (KicKerBNU) [inline src/cdn/elements/tabs.css:50]: tabs seems to have same problem with inverted directions

File: src/cdn/elements/tooltips.css
Comment (KicKerBNU) [inline src/cdn/elements/tooltips.css:33]: im not sure the directions of tooltip should be inverted

File: src/cdn/helpers/positions.css
Comment (KicKerBNU) [inline src/cdn/helpers/positions.css:9]: not sure if left should be displayed on right on RTL

File: src/cdn/helpers/positions.css
Comment (KicKerBNU) [inline src/cdn/helpers/positions.css:13]: not sure if right should be displayed on left on RTL

**Review by KicKerBNU — COMMENTED** [review]
Aditional considerations:

- expansions not showing the text as expected
- the code button used on beercss site show some bugs on start and end of classes

Final consideration: Im not familiar with RTL language, the comments bellow are based on tests using LTR-RTL Chrome plugin

**Comment by coffeeandwork** [comment]
@KicKerBNU thank you for taking the time to review my code. 

To be sure, in some ways, if we do not add the code in this PR to the beercss library, anyone using the beercss library who needs to support RTL languages will have to have `if (RTL Mode) use 'left' css class; else 'right' css class;` in many places in their code. I may propose to you that when a ui library claims to support RTL, it means that whoever uses that library does not have to have such if-statements all over their code because the ui library handles it for them.

You are right in pointing out that with this PR, generally anything we mark as "left" via css classes, will show up on the right in RTL mode and things we mark as "right"  via css class names, will show up on the left in RTL mode.  I propose we have two possible options here:

1. We can say that most of the developers (even those whose native language happens to be an RTL language), will be coding with an LTR mindset because html/css/js can be thought of as english-derived programming languages (english is an LTR language). Thus, we should provide names that make sense in the LTR context (so `"left"` css class means the leftward physical direction in the viewport), with the understanding that everything is the opposite (on the horizontal axis) in RTL.
2. Recognizing this problem with using `left` and `right` being confusing when considering the RTL/LTR languages together, the CSS standard has, for some time now, supported the notion of `inline-start` and `inline-end` where in LTR, `inline-start` is `left` and `inline-end` is `right` and the opposite in RTL. We could use this `inline-start/end` terminology.

I would vote for option (1) above because I think most developers may feel more comfortable with `left` and `right` than with `inline-start` and `inline-end`, and I suspect that only a small number of developers worry about RTL language support, optimizing for the majority users' experience seems to have the best ROI. I am happy to do (2) as well if others' feel strongly about it.

I look forward to receiving your feedback.

**Comment by KicKerBNU** [comment]
Well lets stay with the option one, maybe we can add some RTL documentation and grab some feedback in the future from people who uses RTL languages to improve it, nice job @coffeeandwork.

**Review by KicKerBNU — APPROVED** [review]

**Comment by coffeeandwork** [comment]
Thank you everyone for your thoughts/feedback/review. @leonardorafael can we merge this PR soon? Looks like its ready to go and we have multiple approvals.

**Pull request merged**
